src/sentry/models/group.py

```diff
@@ -646,3 +658,32 @@ def times_seen_pending(self) -> int:
     @times_seen_pending.setter
     def times_seen_pending(self, times_seen: int):
         self._times_seen_pending = times_seen
+
+    def get_type_category_mapping(self):
+        return {
+            GroupType.ERROR: IssueCategory.ERROR,
+            GroupType.PERFORMANCE_N_PLUS_ONE: IssueCategory.PERFORMANCE,
+            GroupType.PERFORMANCE_SLOW_SPAN: IssueCategory.PERFORMANCE,
+        }
+
+    def get_category_type_mapping(self):
+        category_type_mapping = {}
+
+        for type, category in self.get_type_category_mapping().items():
+            if category_type_mapping.get(category):
+                category_type_mapping[category].append(type)
+            else:
+                category_type_mapping[category] = [type]
+
+        return category_type_mapping
```

Comment:
Would it make sense to replace these methods with a module level mapping:

```
GROUP_TYPE_TO_CATEGORY = {
  GroupType.ERROR: IssueCategory.ERROR,
  GroupType.PERFORMANCE_N_PLUS_ONE: IssueCategory.PERFORMANCE,
  GroupType.PERFORMANCE_SLOW_SPAN: IssueCategory.PERFORMANCE,
}

GROUP_CATEGORY_TO_TYPE = {
  IssueCategory.ERROR: GroupType.ERROR,
  IssueCategory.PERFORMANCE: GroupType.PERFORMANCE_N_PLUS_ONE, 
  IssueCategory.PERFORMANCE: GroupType.PERFORMANCE_SLOW_SPAN:,
}
```

Reply:
`get_category_type_mapping` returns a dict where the key is an `IssueCategory` and the value is a list of `GroupTypes` that belong to that category to be used in search later on. I could put the return value of  `get_type_category_mapping` in a const though if you anticipate needing that outside of the `Group` class.

Reply:
I think it will be helpful to define these outside of the `Group` class, since we will need `GROUP_CATEGORY_TO_TYPE` to help with search. When we make a search, users can make queries like `category:performance`. This translates to a postgres query like `type in (1, 2, 3, 4, 5)`. To build that query, we'd do something like `group_types = [group_type.value for group_type in GROUP_CATEGORY_TO_TYPE[category]]`